src/main/java/io/debezium/connector/spanner/task/TaskStateChangeEventProcessor.java

```diff
@@ -55,6 +61,35 @@ public TaskStateChangeEventProcessor(int queueCapacity, TaskSyncContextHolder ta
         this.metricsEventPublisher = metricsEventPublisher;
     }
 
+    private Thread createEventQueueingThread() {
+        Thread thread = new Thread(() -> {
```

Comment:
As you touched thread creation here and in other places, could you please use `io.debezium.util.Threads.threadFactory` to create new threads?

Reply:
Hi Jiri, thanks for the feedback! Similar to the logging comment for Chris, could I open a JIRA to use ThreadFactory instead, and address it later when I have more time? I'd have to start another round of testing to make sure ThreadFactory is working as expected.